pulsar/consumer_partition.go

```diff
@@ -1062,16 +1065,19 @@ func getPreviousMessage(mid trackingMessageID) trackingMessageID {
 }
 
 func (pc *partitionConsumer) Decompress(msgMeta *pb.MessageMetadata, payload internal.Buffer) (internal.Buffer, error) {
+	pc.providersMutex.Lock()
```

Comment:
Are there any overhead concerns for now having to grab a new lock each time we receive a message? 

cc @merlimat

Reply:
@fishpenguin , @cckellogg ,
I'm thinking an [RWMutex](https://golang.org/pkg/sync/#RWMutex) would generate less contention.  The majority of accesses require the read lock; the write lock is only required when initialising the provider.